### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by rasendubi**
Initial implementation of Rust SDK.

This PR is best reviewed by commits.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+[package]
+name = "eppo"
+version = "0.1.0"
```
Comment (leoromanovsky) [inline Cargo.toml:3]: Good for now - will bump to `1.0.0` when ready for release. Is that appropriate convention in rust?

File: Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+edition = "2021"
+description = "Eppo SDK for Rust"
+homepage = "https://docs.geteppo.com/sdks/server-sdks/rust"
+repository = "https://github.com/Eppo-exp/rust-sdk"
+license = "MIT"
```
Comment (leoromanovsky) [inline Cargo.toml:8]: 👍

File: src/config.rs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    pub fn assignment_logger(
+        &mut self,
+        assignment_logger: impl AssignmentLogger + Send + Sync + 'a,
+    ) -> &mut Self {
+        self.assignment_logger = Box::new(assignment_logger);
+        self
+    }
+
+    /// Default base URL for API calls.
+    pub const DEFAULT_BASE_URL: &str = "https://fscdn.eppo.cloud/api";
```
Comment (leoromanovsky) [inline src/config.rs:42]:
```suggestion
    pub const DEFAULT_BASE_URL: &'static str = "https://fscdn.eppo.cloud/api";
```

File: src/assignment_logger.rs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+#[derive(Debug, Serialize, Deserialize)]
+pub struct AssignmentEvent {
+    pub experiment: String,
+    pub variation: String,
+    pub subject: String,
+    pub subject_attributes: SubjectAttributes,
+    pub timestamp: String,
+    pub feature_flag: String,
+    pub allocation: String,
+}
```
Comment (leoromanovsky) [inline src/assignment_logger.rs:14]: This definition is almost the desired shape just missing the `metaData` field which is a generic container. 

In the python implementation is contains:

```
"metaData": {"sdkLanguage": "python", "sdkVersion": __version__},
```

No problem! I apologize if the design spec didn't make clear but I  wanted to mention again that the UFC migration code in the javascript and python code is not yet merged to main. 

Two SDKs have been converted to the forthcoming “UFC” format are:

* https://github.com/Eppo-exp/python-sdk/pull/28
* https://github.com/Eppo-exp/js-client-sdk-common/pull/45

Sorry again if this caused confusion - they should be merged to main shortly to make reading them easier.

**Review by leoromanovsky — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+name: Cargo Build & Test
```
Comment (leoromanovsky) [inline .github/workflows/ci.yml:1]: Did you expect this file to run in the PR?

Comment (rasendubi) [reply .github/workflows/ci.yml:1]: It needs to be merged to affect PRs. So far it runs on my repo: https://github.com/rasendubi/eppo-rust-sdk/actions/runs/8803705195/job/24162446849

File: src/config.rs (hunk 3 above)
Comment (rasendubi) [reply src/config.rs:42]: Any reason you prefer this more verbose version?

File: src/assignment_logger.rs (hunk 4 above)
Comment (rasendubi) [reply src/assignment_logger.rs]: Good catch! I added this field and also added missing `camelCase` transformation

File: src/config.rs (hunk 3 above)
Comment (leoromanovsky) [reply src/config.rs:42]: The Rust compiler showed this warning and even suggested this fix when I ran:

```
cargo fix --lib -p eppo
```

Comment (rasendubi) [reply src/config.rs:42]: Which version of rust/cargo do you use?

Comment (rasendubi) [reply src/config.rs:42]: Nevermind. Found it

Comment (leoromanovsky) [reply src/config.rs:42]: Just installed it yesterday!

```
cargo --version
cargo 1.77.2
```

File: Cargo.toml (hunk 1 above)
Comment (schmit) [reply Cargo.toml:3]: I think we want to publish this as version 3.0.0 once fully ready to indicate that this version of the SDK is using the UFC

**Pull request merged**